--- frandom-0.8/frandom.c~	2006-06-28 15:39:07.000000000 +0200
+++ frandom-0.8/frandom.c	2006-06-28 16:15:59.000000000 +0200
@@ -60,15 +60,15 @@
 MODULE_DESCRIPTION("Fast pseudo-random number generator");
 MODULE_LICENSE("GPL");
 MODULE_AUTHOR("Eli Billauer");
-MODULE_PARM(frandom_major,"i");
+module_param(frandom_major, int, FRANDOM_MAJOR);
 MODULE_PARM_DESC(frandom_major,"Major number of /dev/frandom and /dev/erandom");
-MODULE_PARM(frandom_minor,"i");
+module_param(frandom_minor, int, FRANDOM_MINOR);
 MODULE_PARM_DESC(frandom_minor,"Minor number of /dev/frandom");
-MODULE_PARM(erandom_minor,"i");
+module_param(erandom_minor, int, ERANDOM_MINOR);
 MODULE_PARM_DESC(erandom_minor,"Minor number of /dev/erandom");
-MODULE_PARM(frandom_bufsize,"i");
+module_param(frandom_bufsize, int, 256);
 MODULE_PARM_DESC(frandom_bufsize,"Internal buffer size in bytes. Default is 256. Must be >= 256");
-MODULE_PARM(frandom_chunklimit,"i");
+module_param(frandom_chunklimit, int, 0);
 MODULE_PARM_DESC(frandom_chunklimit,"Limit for read() blocks size. 0 (default) is unlimited, otherwise must be >= 256");
 
 struct frandom_state
